salt/fileclient.py

```diff
@@ -462,6 +462,18 @@ def get_url(self, url, dest, makedirs=False, saltenv='base',
         '''
         url_data = urlparse(url)
 
+        # If dest is a directory, rewrite dest with filename
+        if os.path.isdir(dest) or dest.endswith(('/', '\\')):
+            if url_data.query or len(url_data.path) > 1 and not url_data.path.endswith('/'):
+                strpath = url.split('/')[-1]
+            else:
+                strpath = 'index.html'
```

Comment:
Why are we adding an ``index.html`` to the URL being queried? We can't assume this path will exist for remote URLs?

Furthermore, why are we messing with the URL path at all, when this pull request is only supposed to allow for the destination to be different? It seems to me that, if you have a URL which contains querystring data in it, then it should be up to you to specify the full destination path, instead of trying to get cute and determine it based on the URL?